packages/viewport/src/with-viewport-match.native.js

```diff
@@ -27,24 +22,28 @@ import { store } from './store';
  *
  * ```jsx
  * function MyComponent( { isMobile } ) {
- * 	return (
- * 		<div>Currently: { isMobile ? 'Mobile' : 'Not Mobile' }</div>
- * 	);
+ *     return (
+ *         <div>Currently: { isMobile ? 'Mobile' : 'Not Mobile' }</div>
+ *     );
  * }
  *
  * MyComponent = withViewportMatch( { isMobile: '< small' } )( MyComponent );
  * ```
  *
  * @return {Function} Higher-order component.
  */
-const withViewportMatch = ( queries ) =>
-	createHigherOrderComponent(
+const withViewportMatch = ( queries ) => {
+	const queryEntries = Object.entries( queries );
+	return createHigherOrderComponent(
 		withSelect( ( select ) => {
-			return mapValues( queries, ( query ) => {
-				return select( store ).isViewportMatch( query );
-			} );
+			return Object.fromEntries(
+				queryEntries.map( ( [ key, query ] ) => {
+					return [ key, select( store ).isViewportMatch( query ) ];
```

Comment:
I think I'm leaning towards keeping it as-is. Do you see any potential big performance improvements from that change?

Reply:
Not really, it's just a double map lookup, equivalent to `registry[ store ][ method ]`. On the other hand, `Object.entries` is in a different ballpark, it would allocate and deallocate memory on each loop iteration.